legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-generation/legend-engine-xt-relationalStore-pure/legend-engine-xt-relationalStore-sqlDialectTranslation-pure/src/main/resources/core_external_store_relational_sql_dialect_translation/defaults/sqlDialectDefaultsTests.pure

```diff
@@ -1040,3 +1053,40 @@ function <<test.Test>> meta::external::store::relational::sqlDialectTranslation:
     '    103 < 104\n' +
     ')', $dialect->executeNodeProcessor(^SubqueryExpression(query = ^Query(queryBody = $q1)), $state, $config));
 }
+
+
+function <<test.Test>> meta::external::store::relational::sqlDialectTranslation::defaults::tests::testTableSubqueryProcessing(): Boolean[1]
+{
+  let dialect = testDialect();
+  let state = testSqlDialectTranslationState();
+  let config = testSqlDialectTranslationConfig(false);
+
+  let q1 = ^QuerySpecification(
+    select = ^Select(selectItems = [^SingleColumn(expression = ^ArithmeticExpression(left = literal(101), right = literal(102), type = ArithmeticType.ADD))]),
+    from = ^Table(name = ^QualifiedName(parts = ['t'])),
+    where = ^ComparisonExpression(left = literal(103), right = literal(104), operator = ComparisonOperator.LESS_THAN)
+  );
+  assertEquals('(SELECT 101 + 102 FROM t WHERE 103 < 104)', $dialect->executeNodeProcessor(^TableSubquery(query = ^Query(queryBody = $q1)), $state, $config));
+}
+
+function <<test.Test>> meta::external::store::relational::sqlDialectTranslation::defaults::tests::testTableSubqueryPrettyFormatProcessing(): Boolean[1]
+{
+  let dialect = testDialect();
+  let state = testSqlDialectTranslationState();
+  let config = testSqlDialectTranslationConfig(true);
+
+  let q1 = ^QuerySpecification(

```

Comment:
it would be clearer to write the input in the sql dialect rather than object construction